src/main/java/com/salesforce/storm/spout/sideline/trigger/example/ZookeeperWatchTrigger.java

```diff
@@ -0,0 +1,417 @@
+        }
+
+        if (triggerEvent.getType().equals(SidelineType.START)) {
+            logger.info("Starting sideline request {} from event {}", sidelineRequest, triggerEvent);
+            sidelineController.startSidelining(sidelineRequest);
+        }
+
+        if (triggerEvent.getType().equals(SidelineType.STOP)) {
+            logger.info("Stopping sideline request {} from event {}", sidelineRequest, triggerEvent);
+            sidelineController.stopSidelining(sidelineRequest);
+        }
+
+        // Write the trigger event back to its path and flip the processed bit to true
+        curatorHelper.writeJson(path, new TriggerEvent(
+            triggerEvent.getType(),
+            triggerEvent.getData(),
+            triggerEvent.getCreatedAt(),
+            triggerEvent.getCreatedBy(),
+            triggerEvent.getDescription(),
+            // Explicit set this as processed
+            true,
+            // Update the updated at date to right now
+            new Date()
+        ));
+    }
+
+    /**
+     * Build a sideline request from a trigger event (what came from Zookeeper).
+     * @param triggerEvent The trigger event
+     * @return A sideline request that can be used to start of stop sidelining
+     */
+    private SidelineRequest buildSidelineRequest(final TriggerEvent triggerEvent) {
+        try {
+            final FilterChainStep step = filterChainStepBuilder.build(triggerEvent.getData());
+
+            final SidelineRequest sidelineRequest = new SidelineRequest(
+                generateSidelineRequestIdentifier(triggerEvent, step),
+                step
+            );
+
+            logger.info("Creating a sideline request with id {} and step {}", sidelineRequest.id, sidelineRequest.step);
+
+            return sidelineRequest;
+        } catch (NoSuchAlgorithmException | UnsupportedEncodingException ex) {
+            logger.error("Unable to generate an identifier for this request, cowardly refusing to proceed! {}", ex);
+            return null;
+        }
+    }
+
+    /**
+     * Using a filter chain step, make some JSON out of it and then hash it to create an idempotent identifier.
+     * @param triggerEvent Trigger event that contains metadata about the request
+     * @param step The FilterChainStep that we are going to generate the request from
+     * @return A sideline request identifier for the filter chain step
+     * @throws NoSuchAlgorithmException Your java install is whack yo, it's missing MD5, for realz???
+     */
+    private SidelineRequestIdentifier generateSidelineRequestIdentifier(
+        final TriggerEvent triggerEvent,
+        final FilterChainStep step
+    ) throws NoSuchAlgorithmException, UnsupportedEncodingException {
+        final String json = gson.toJson(step);
+
+        final StringBuilder identifier = new StringBuilder(
+            DatatypeConverter.printHexBinary(MessageDigest.getInstance("MD5").digest(json.getBytes("UTF-8")))
```

Comment:
probably ok, but isnt MD5 considered dead?

Reply:
Dead in what way?

Reply:
I think mostly for crypto / being able to easily force collisions.  In this context it's probably safe to use.

Reply:
It's simple and fast and should be fairly accurate here so I'm not horribly worried. We can always make it configurable if it becomes an issue.